src/core/hooks.ts

```diff
@@ -1,7 +1,3 @@
-import type { Grammar, TokenName } from '../types';
-import type { HookState } from './hook-state';
-import type { TokenStream } from './token';
-
 export class Hooks {
 	// eslint-disable-next-line func-call-spacing
 	private _all = new Map<string, ((env: unknown) => void)[]>();
```

Comment:
What if we switch to native private properties and methods? I think TS will still correctly infer the property visibility. Or is there a reason to stick with the `_` convention?

Reply:
`_all` indicates that it's private, but is still readable by any code doing some weird low-level thing. That said, it doesn't seem to be read by any other part of the codebase, so we could, though it also doesn't provide much of a benefit so I don't think it's particularly important, so I'm inclined to leave it as-is.